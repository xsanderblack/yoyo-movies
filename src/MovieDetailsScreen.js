import React from 'react';
import MoviePoster from './MoviePoster';

const MovieDetailsScreen = ({
  release_date,
  production_countries,
  runtime,
  vote_average,
  overview,
  homepage,
  poster_path,
  title
}) => (
  <div>
    <MoviePoster size='full' path={poster_path} alt={title} />
    <ul>
      {release_date && <li>Release date: {release_date}</li>}
      {production_countries &&
        <li>Countries: {production_countries.map((country) =>
          <p key={country.iso_3166_1}>{country.name}</p>
        )}
        </li>
      }
      {runtime && <li>Run time: {runtime}</li>}
      {vote_average && <li>Rating: {vote_average}</li>}
      {homepage && <li>Home page: {homepage}</li>}
    </ul>
    <p>{overview}</p>
  </div>
);

export default MovieDetailsScreen;
