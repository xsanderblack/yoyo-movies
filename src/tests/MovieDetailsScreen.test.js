import React from 'react';
import ReactDOM from 'react-dom';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import MovieDetailsScreen from '../MovieDetailsScreen';

it('MovieDetailsScreen renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MovieDetailsScreen />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should render MovieDetailsScreen correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<MovieDetailsScreen />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
