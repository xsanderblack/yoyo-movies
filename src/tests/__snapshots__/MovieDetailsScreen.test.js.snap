// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieDetailsScreen correctly 1`] = `
<div>
  <MoviePoster
    size="full"
  />
  <ul />
  <p />
</div>
`;
